npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -800,7 +800,7 @@ def restore_packages_name(updated_lockfile_content, parsed_updated_lockfile_cont
 
         sig do
           params(
-            current_name: String,
+            current_name: T.any(String, NilClass),
```

Comment:
I have a couple of points:
1. The more idiomatic way to write this expression is `T.nilable(String)`
2. While this change solves the Sorbet error, I believe it's actually a code defect. See #10158 for my attempted solution

Reply:
Thanks @JamieMagee and @kbukum1 , Jamie, is the recommendation to get the [PR](https://github.com/dependabot/dependabot-core/pull/10158) over the line then?

Reply:
@abdulapopoola I believe so. I just haven't had capacity to write the unit test. @kbukum1 you're welcome to take over my branch if you'd like.

Reply:
@kbukum1 , can you please help add the unit test and we can get that PR merged and close this one. Thanks!

Reply:
Just I want to mention, 

When I tried to rerun this error for the repositories which had problem. I didn't get error. It is running without errror and when I check the current_name it is always a name generated by dependabot. So I thought this is happening at first time and then dependabot is able to generate current name then there is no error happening. Note that seems like this is happening when there is no name in package.json. 

What do you think @JamieMagee ? 
CC: @abdulapopoola

Reply:
I am going to add unit tests to your PR. As for this PR we may not need to make changes on Sorbet that. Going to close that one if everything is ok with your PR. Is that ok @JamieMagee